### PR TITLE
Optimize multisearch method

### PR DESCRIPTION
**Pull request by xiaojun207**
1、Optimize multisearch method (in pkg/core/multi_search.go)

### CODE REVIEW

**Comment by hengfeiyang** [comment]
Thank you for your work, Can you add some tests for you added `isMatchIndex` in `multi_search_test.go`

We should coverage your code.

**Comment by xiaojun207** [comment]
> 
Ok，good idea，I  added the test for isMatchIndex in multi_search_test.go

File: pkg/core/multi_search.go [hunk 1]
```diff
@@ -30,31 +30,58 @@ import (
...
 )
 
+// isMatchIndex("abc", "a")  false
+// isMatchIndex("abc", "a*") true
+// isMatchIndex("abc", "*bc") true
+// isMatchIndex("abc", "bc") false
+// isMatchIndex("abc", "abc") true
+func isMatchIndex(zincIndexName, indexName string) bool {
+	// the old code: indexName == "" || (indexName != "" && strings.HasPrefix(zincIndexName, indexName[:len(indexName)-1]))
+
```
Comment (hengfeiyang) [inline pkg/core/multi_search.go:40]: I think we can delete it, i known your mean, but we don't need keep this in code for main branch.

File: pkg/core/multi_search.go [hunk 2]
```diff
@@ -30,31 +30,58 @@ import (
...
 func MultiSearch(indexNames []string, query *meta.ZincQuery) (*meta.SearchResponse, error) {
 	var mappings *meta.Mappings
 	var analyzers map[string]*analysis.Analyzer
 	var readers []*bluge.Reader
 	var shardNum int
-	indexMap := make(map[string]struct{})
 
 	timeMin, timeMax := timerange.Query(query.Query)
 	for _, index := range ZINC_INDEX_LIST.List() {
+		isMatched := false
```
Comment (hengfeiyang) [inline pkg/core/multi_search.go:68]: We can move it out of for, then we will just have one bool variable.

File: pkg/core/multi_search.go [hunk 3]
```diff
@@ -30,31 +30,58 @@ import (
...
 		for _, indexName := range indexNames {
-			if _, ok := indexMap[index.Name]; ok {
-				continue
+			isMatched = isMatchIndex(index.Name, indexName)
+			if isMatched {
+				break
+			}
+		}
+		if isMatched {
```
Comment (hengfeiyang) [inline pkg/core/multi_search.go:75]: We can set `if !isMatched` then break, then we will reduce one nest.

Comment (xiaojun207) [reply pkg/core/multi_search.go:73]: If set if !isMatched then break, The following ZincIndexNames haven't check match yet，I think we need check left in “ MultiSearch method”。What about you?

Comment (hengfeiyang) [reply pkg/core/multi_search.go:73]: Sorry, i made a mistake, i mean we should if `!isMatched then `continue`, then we will reduce one nest, the code format will be better.

Comment (xiaojun207) [reply pkg/core/multi_search.go]: Yes ，its better

**Pull request merged**